src/app/_validators/sitting-records-form-validator.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { AbstractControl, FormControl, FormGroup } from '@angular/forms';
+import { CustomValidators } from './sitting-records-form-validator';
+
+describe('CustomValidators', () => {
+
+    describe('validateDateFormat', () => {
+        let control: FormGroup;
+
+        beforeEach(() => {
+            control = new FormGroup({
+                dateDay: new FormControl(''),
+                dateMonth: new FormControl(''),
+                dateYear: new FormControl('')
+            });
+            control.get('dateDay')?.markAsDirty();
+            control.get('dateMonth')?.markAsDirty();
+            control.get('dateYear')?.markAsDirty();
+        });
+
+        it('should return date_invalid error when day, month or year is invalid', () => {
+            control.patchValue({ dateDay: '1', dateMonth: '13', dateYear: '2023' });
+            control.get('dateDay')?.markAsTouched();
```

Comment:
Do we need markAsTouched for dateDay, dateMonth and dateYear as we have set them to markAsDirty in the beforeEach() and 'required' validator is set on them?

Reply:
Both are required for different parts of the validator logic, removing this would cause tests to fail/